test/s3select_test_json_parser.cpp

```diff
@@ -656,13 +952,24 @@ int sax_row_count(const char *in, std::vector<std::string>& from_clause)
 	return -1;
 }
 
-TEST(TestS3selectJsonParser, sax_vs_dom)
-{/* the dom parser result is compared against sax parser result
-	ASSERT_EQ( compare_results(TEST2) ,0); // Commenting as it is not implemented in the server side
-	ASSERT_EQ( compare_results(TEST3) ,0);
-	ASSERT_EQ( compare_results(TEST4) ,0);*/
-}
+int run_json_query(const char* json_query, std::string& json_input,std::string& result)
+{//purpose: run single-chunk json queries
+
+  s3selectEngine::s3select s3select_syntax;
+  int status = s3select_syntax.parse_query(json_query);
+  if (status != 0)
+  {
+    std::cout << "failed to parse query " << s3select_syntax.get_error_description() << std::endl;
+    return -1;
+  }
 
+  s3selectEngine::json_object json_query_processor(&s3select_syntax);
+  status = json_query_processor.run_s3select_on_stream(result, json_input.data(), json_input.size(), json_input.size());
+  status = json_query_processor.run_s3select_on_stream(result, 0, 0, json_input.size());
```

Comment:
the second call to `run_s3select_on_stream` might overrides the previous result.

Reply:
what about this?